### PR TITLE
fix: broken mdx parsing of accordions

### PR DESCRIPTION
**Pull request by abvthecity**
Fixes issue introduced in https://github.com/fern-api/fern-platform/pull/1683

- accordions are not parsed correctly by value-to-estree
- moved more utilities into fern-docs-mdx
- added tests and comments

### CODE REVIEW

**Review by RohinBhargava — APPROVED** [review]

File: packages/ui/fern-docs-mdx/src/mdx-utils/unknown-to-estree-expression.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { isPlainObject } from "@fern-api/ui-core-utils";
+import type { Expression, ExpressionStatement } from "estree";
+import { valueToEstree, type Options as ValueToEstreeOptions } from "estree-util-value-to-estree";
+import { isMdxJsxAttributeValueExpression } from "./is-mdx-jsx-attr.js";
+
+export function unknownToEstreeExpression(value: unknown, options?: ValueToEstreeOptions): Expression {
```
Comment (abvthecity) [inline packages/ui/fern-docs-mdx/src/mdx-utils/unknown-to-estree-expression.ts:6]: This is the fix

File: packages/ui/fern-docs-mdx/src/mdx-utils/unknown-to-mdx-jsx-attr.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    // generate a new expression program
+    return {
+        type: "mdxJsxAttributeValueExpression",
+        value: "__expression__",
+        data: {
+            estree: {
+                type: "Program",
+                sourceType: "module",
+                body: [{ type: "ExpressionStatement", expression: unknownToEstreeExpression(value) }],
```
Comment (abvthecity) [inline packages/ui/fern-docs-mdx/src/mdx-utils/unknown-to-mdx-jsx-attr.ts:23]: This is where the fix is being used

File: packages/ui/app/src/mdx/plugins/rehypeFernComponents.ts [hunk 3]
```diff
@@ -260,62 +274,40 @@ function transformSteps(
...
-
-        // expression attributes are not supported
-    });
-
-    if (child.children.length > 0) {
-        props.children = {
-            type: "mdxJsxFlowElement",
-            name: "Fragment",
-            attributes: [],
-            children: child.children.map((child) => (child.type === "text" ? h("p", {}, child.value) : child)),
```
Comment (abvthecity) [inline packages/ui/app/src/mdx/plugins/rehypeFernComponents.ts:302]: This is where it broke

File: packages/ui/app/src/mdx/plugins/rehypeFernComponents.ts [hunk 4]
```diff
@@ -183,7 +191,13 @@ function transformAccordionGroup(
     const child = {
         type: "mdxJsxFlowElement" as const,
         name: "AccordionGroup",
-        attributes: [toAttribute("items", items.map(collectProps)), ...node.attributes],
+        attributes: [
+            unknownToMdxJsxAttribute(
+                "items",
+                items.map((item) => hastMdxJsxElementHastToProps(item).props),
```
Comment (abvthecity) [inline packages/ui/app/src/mdx/plugins/rehypeFernComponents.ts:197]: This is where the fix is consumed

**Pull request merged**